### ISSUE TITLE
TypeError: axe.run arguments are invalid

### ISSUE THREAD

**Issue by martinezvazquez122** [dequelabs/axe-core#941]
Hello! I am having some issues when using the Axe.run() method when running it on my Stencil project. I am getting the following message: TypeError: axe.run arguments are invalid. Here are the details:

-axe-core version: 3.0.2 (also tried 3.0.3)

Browser and Assistive Technology versions:
stencil/core: 0.9.0

For Tooling issues:
- Node version: 8.9.4
- Platform:  Mac

<pre><code>
import { TestWindow } from '@stencil/core/testing';
import { MyComponent } from './mycomponent';
import * as Axe from 'axe-core';

describe('my-component', () => {
  let window;
  let element;

  beforeEach(async () => {
    window = new TestWindow();
    element = await window.load({
      components: [MyComponent],
      html: '<my-component"></my-component>',
    });
  });

  it('should have accessibility violations depending on values specified', async () => {
    const spy = jest.spyOn(Axe, 'run');

    const options: Axe.RunOptions = {
      runOnly: {
        type: 'tags',
        values: ['wcag2a', 'wcag2aa'],
      },
    };

    const context: Axe.ElementContext = element;
    const test = await Axe.run(element, options);

    expect(test).not.toBeNull();

    Axe.run(context).then(results => {
      expect(results.violations.length).toEqual(4);
    });

    Axe.run(options).then(results => {
      expect(results.violations.length).toEqual(2);
    });

    Axe.run().then(results => {
      expect(results.violations.length).toEqual(4);
    });

    expect(spy).toHaveBeenCalledTimes(3);
  });
});
</code></pre>

The code above works fine but if I try to use more than one param in the run call it gives me TypeError: axe.run arguments are invalid.
For example:

<pre><code>
    Axe.run(context, options).then(results => {
      expect(results.violations.length).toEqual(2);
    });
</code></pre>

Thank you in advance.

**Comment by marcysutton** [comment]
I'm seeing issues with multiple calls to axe.run in the same test, at least in Jasmine and headless Chrome. To try and fully reproduce this issue, can you please provide the complete source code so we can install and run the tests ourselves?

**Comment by martinezvazquez122** [comment]
Hello! Thank you for the reply. Please see a sample project at https://github.com/martinezvazquez122/stencil-component-starter.

- Checkout the branch axe-core_for_stencil

- Run npm install

- Then the test file is located at /src/components/my-component/my-component.spec.ts.

- You can run that specific file by doing npm  run test my-component.spec.ts

**Comment by martinezvazquez122** [comment]
If you comment out the last piece of the test in the my-component.spec.ts file, the test should fail.

**Comment by marcysutton** [comment]
@martinezvazquez122 it's a problem with the context–you have to pass a real DOM node to axe.run. The first example kinda works because it falls back to the document. But when you try to use multiple arguments, axe-core chokes when trying to sort them out. When you log `element`, it comes back with [HTMLUnknownElement](https://developer.mozilla.org/en-US/docs/Web/API/HTMLUnknownElement), which doesn't contain the necessary properties and methods axe-core needs to function.

I'm going to close this since it isn't really an axe-core problem, but more a problem with how the Stencil tests are structured. If you want to discuss further, come find us on Gitter: https://gitter.im/dequelabs/axe-core

**Comment by WilcoFiers** [comment]
Yeah, this doesn't work because Axe currently requires there to be a global window object before it is loaded. We've used jsdom-global in the past for this. It is something I'd like to change - have Axe pull everything from the context, instead of having it require a number of global variables.

**Comment by marcysutton** [comment]
That would probably also help with Cypress! #755

**Comment by dylanb** [comment]
Solution found

```
describe('Inject axe-core and run it', function() {
  it('Runs axe-core', function() {
    cy.visit('your page')
      .window().then((win) => {
        var window = win;
        var document = win.document;
        var axe = require('axe-core');
        return axe.run(window.document);
      }).then((results) => {
        console.log(results);
        expect(results.violations.length).to.equal(0);
      })
  })
})
```
